source/common/modules/markdown-editor/hooks/note-preview.js

```diff
@@ -0,0 +1,95 @@
+const tippy = require('tippy.js').default
+const { ipcRenderer } = require('electron')
+const { DateTime } = require('luxon')
+/**
+ * A hook for displaying link tooltips which display metadata
+ * and content of a file
+ *
+ * @param   {CodeMirror}  cm  The instance to attach to
+ */
+module.exports = (elem) => {
+  elem.getWrapperElement().addEventListener('mousemove', (event) => {
+    let a = event.target
+
+    // Only for note links
+    if (!a.classList.contains('cm-zkn-link')) {
+      return
+    }
+
+    // If there's already a tippy on the instance, don't re-render it
+    if (a.hasOwnProperty('_tippy')) {
+      return
+    }
+    // TODO: Translate!
+    // Initialise displayed attributes to 'null'
+    let title = 'null'
+    let content = 'null'
+    let wordCount = 'null'
+    let time = 'null'
+    // Create a tippy. This will display the loading values
+    let tooltip = tippy(a, {
+      content: 'Searching For File...',
+      allowHTML: true, // Obviously
+      interactive: true,
+      placement: 'top-start', // Display at the beginning of the anchor
+      appendTo: document.body, // anchor
+      showOnCreate: true, // Immediately show the tooltip
+      arrow: false // No arrow for these tooltips
+    })
+
+    // Find the file's absolute path
+    ipcRenderer.invoke('application', { command: 'file-find-and-return', payload: a.innerText })
+      .then((descriptorWithContent) => {
```

Comment:
It would still be great if you could move all that logic to a separate function which you call here (either by simply passing it as a reference, or by calling it from within the callback).